presto-native-execution/presto_cpp/main/PrestoExchangeSource.h

```diff
@@ -100,13 +105,43 @@ class PrestoExchangeSource : public velox::exec::ExchangeSource {
       uint32_t maxBytes,
       uint32_t maxWaitSeconds) override;
 
-  static std::shared_ptr<ExchangeSource> create(
+  // HTTP connection pool for a specific endpoint with its associated event
+  // base.  All the operations on the SessionPool must be performed on the
+  // corresponding EventBase.
+  struct ConnectionPool {
+    folly::EventBase* eventBase;
+    std::unique_ptr<proxygen::SessionPool> sessionPool;
+  };
+
+  class ConnectionPools {
+   public:
+    ~ConnectionPools() {
+      destroy();
+    }
+
+    const ConnectionPool& get(
+        const proxygen::Endpoint& endpoint,
+        folly::IOThreadPoolExecutor* ioExecutor);
+    void destroy();
+
+   private:
+    folly::Synchronized<folly::F14FastMap<
+        proxygen::Endpoint,
+        ConnectionPool,
+        proxygen::EndpointHash,
+        proxygen::EndpointEqual>>
+        pools_;
+  };
```

Comment:
If feels very weird that ConnectionPools is declared inside of PrestoExchangeSource class.
Because it is actually owned by someone else.
It made me confused for a while.
Should we consider declaring ConnectionPools in the global namespace?

Reply:
I put it here to make it clear these connection pools are used by exchange source (not HTTP server or anything else)

Reply:
@Yuhta 
We could add a comment for this class.
Usually classes declared within other classes, when the objects of the internal classes are created/managed/used by the external class.
Not our case.